hypothesis-python/src/hypothesis/extra/pandas/impl.py

```diff
@@ -147,7 +148,7 @@ def do_draw(self, data):
         dtype = infer_dtype_if_necessary(
             dtype=self.dtype, values=result, elements=self.elements, draw=data.draw
         )
-        return pandas.Index(result, dtype=dtype, tupleize_cols=False)
+        return pandas.Index(result, dtype=dtype, tupleize_cols=False, name=self.name)
```

Comment:
```suggestion
        return pandas.Index(result, dtype=dtype, tupleize_cols=False, name=data.draw(self.name))
```

Since this is now a strategy, we'll need to draw a value from it.  (and again below)